app/assets/javascripts/discourse/models/login-method.js.es6

```diff
@@ -45,7 +49,8 @@ const LoginMethod = Ember.Object.extend({
         const width = this.get("frame_width") || 800;
 
         if (name === "facebook") {
-          authUrl = authUrl + "?display=popup";
+          authUrl += authUrl.includes("?") ? "&" : "?";
```

Comment:
It's a bit messy to look for the question mark here. If you take my suggestion above of using an object literal for parameters, you could call the argument to doLogin `queryParameters`. Facebook would set `queryParameters.display`. Finally you'd build the query parameters string from the object.